hudi-client/hudi-client-common/src/main/java/org/apache/hudi/io/FileGroupReaderBasedMergeHandle.java

```diff
@@ -106,7 +171,15 @@ public FileGroupReaderBasedMergeHandle(HoodieWriteConfig config, String instantT
     } else {
       this.cdcLogger = Option.empty();
     }
-    init(operation, this.partitionPath);
+  }
+
+  private void initRecordIndexCallback() {
+    if (this.writeStatus.isTrackingSuccessfulWrites()) {
```

Comment:
The `isTrackingSuccessfulWrites` flag in write status comes from `hoodieTable.shouldTrackSuccessRecords()`, which is true when RLI or partitioned RLI is enabled, we should skip the location tracing for compaction which is redundant.